components/post_view/post_attachment_opengraph/post_attachment_opengraph.tsx

```diff
@@ -18,52 +22,23 @@ const DIMENSIONS_NEAREST_POINT_IMAGE = {
     width: 80,
 };
 
-export default class PostAttachmentOpenGraph extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * The link to display the open graph data for
-         */
-        link: PropTypes.string.isRequired,
-
-        /**
-         * The ID of the current user
-         */
-        currentUserId: PropTypes.string,
-
-        /**
-         * The post where this link is included
-         */
-        post: PropTypes.object,
-
-        /**
-         * The open graph data to render
-         */
-        openGraphData: PropTypes.object,
-
-        /**
-         * Whether or not the server has link previews enabled.
-         */
-        enableLinkPreviews: PropTypes.bool.isRequired,
-
-        /**
-         * Whether or not the user has link previews enabled.
-         */
-        previewEnabled: PropTypes.bool.isRequired,
-
-        /**
-         * Whether or not the image in the OpenGraph preview has been collapsed.
-         */
-        isEmbedVisible: PropTypes.bool,
-
-        toggleEmbedVisibility: PropTypes.func.isRequired,
-
-        actions: PropTypes.shape({
-            editPost: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+export type Props = {
+    postId: string;
+    link: string;
+    currentUserId?: string;
+    post: Post;
+    openGraphData?: OpenGraphMetadata | null;
```

Comment:
Can this just be `openGraphData?: OpenGraphMetadata`? I'm not sure if the selector ever returns null, and the unit tests for this can be updated to pass `undefined` instead.

Up until recently, we've been using null for empty values, but we've since learned undefined is easier to use in most cases.

Reply:
Yeah sure, I'll do that !